src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -999,32 +999,29 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
         /// <returns></returns>
         private TokenValidationResult ValidateToken(string token, JsonWebToken outerToken, string decryptedJwt, TokenValidationParameters validationParameters)
         {
-            var validationParametersCopy = validationParameters;
-            if (validationParametersCopy.ConfigurationManager != null)
+            BaseConfiguration currentConfiguration = null;
+            if (validationParameters.ConfigurationManager != null)
             {
                 try
                 {
-                    // TODO: Use overloads to get rid of the Clone() operation.
-                    var configuration = validationParametersCopy.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
-                    validationParametersCopy = validationParameters.Clone();
-                    validationParametersCopy.Configuration = configuration;
+                    currentConfiguration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).ConfigureAwait(false).GetAwaiter().GetResult();
                 }
                 catch (Exception ex)
                 {
                     // Keep going with the validation as the TokenValidationParameters may have the issuer and signing key set
                     // directly on them.
-                    LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParametersCopy.ConfigurationManager.MetadataAddress, ex.ToString()));
+                    LogHelper.LogInformation(LogHelper.FormatInvariant(TokenLogMessages.IDX10261, validationParameters.ConfigurationManager.MetadataAddress, ex.ToString()));
                 }
             }
 
-            TokenValidationResult tokenValidationResult = decryptedJwt != null ? ValidateJWE(outerToken, decryptedJwt, validationParametersCopy) : ValidateJWS(token, validationParametersCopy);
-            if (validationParametersCopy.ConfigurationManager != null)
+            TokenValidationResult tokenValidationResult = decryptedJwt != null ? ValidateJWE(outerToken, decryptedJwt, validationParameters, currentConfiguration) : ValidateJWS(token, validationParameters, currentConfiguration);
```

Comment:
>decryptedJwt != null ? ValidateJWE(outerToken, decryptedJwt, validationParameters, currentConfiguration) : ValidateJWS(token, validationParameters, currentConfiguration) [](http://example.com/codeflow?start=58&length=169)

Since this appears three times in this method, maybe consider capturing the logic in a method to ensure consistency?

        private TokenValidationResult ValidateJWSOrJWE(string token, JsonWebToken outerToken, string decryptedJwt, TokenValidationParameters validationParameters, BaseConfiguration configuration)
        {
            return decryptedJwt != null ? ValidateJWE(outerToken, decryptedJwt, validationParameters, configuration) : ValidateJWS(token, validationParameters, configuration);
        }

Reply:
@dannybtsai @mafurman we can hold off on any additional cleanup here, if we want, as the changes associated with JsonClaimSet and using SPAN will make some changes in this code.

Reply:
If this will be addressed by JsonClaimSet, I'll hold off on making changes to this.